src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests.cs

```diff
@@ -5043,6 +5043,228 @@ class C<T>
             });
         }
 
+        [Fact]
+        public void TestMainAttributes()
+        {
+            var source = CreateCompilation(@"
+using System;
+
+[main: My(""one"")]
+Console.WriteLine(""Hello World"");
+
+public class MyAttribute : Attribute { public MyAttribute(string name) {} }
+");
+
+            var mainAttrs = source.GetEntryPoint(default).GetAttributes();
+
+            var attribute = Assert.Single(mainAttrs);
+            attribute.VerifyValue(0, TypedConstantKind.Primitive, "one");
+            Assert.Equal(@"MyAttribute(""one"")", attribute.ToString());
+
+            CompileAndVerify(source, expectedOutput: "Hello World");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
```

Comment:
>[ConditionalFact(typeof(CoreClrOnly))] [](http://example.com/codeflow?start=8&length=38)

We should be able to test metadata on all platforms. There is no requirement to test IL for that. In fact symbol validation is more robust.